packages/beacon-node/src/chain/blocks/blockInput/types.ts

```diff
@@ -0,0 +1,165 @@
+import {ForkName} from "@lodestar/params";
+import {ColumnIndex, RootHex, SignedBeaconBlock, Slot, deneb, fulu} from "@lodestar/types";
+
+export type CustodyConfig = {
+  custodyColumns: ColumnIndex[];
+  custodyColumnsIndex: Uint8Array;
+  custodyColumnsLen: number;
+  sampledColumns: ColumnIndex[];
+};
+
+/** Whether a block has been seen and validated for a given BlockInput */
+export enum BlockStatus {
+  /** block has not been seen */
+  MissingBlock,
+  /** block has been seen and validated */
+  HasBlock,
+}
+
+/** Whether DA is required or not for a given BlockInput */
+export enum DARequirement {
+  /** Note: pre-DA, DA is assumed "required" even as no actual DA work is required */
+  Required = "required",
```

Comment:
This is confusing.  If we are going to go with a state machine setup lets model the full range of potential states.  We should have a `preData` enum value and not rely on a note that can get missed by future maintainers
```suggestion
  PreData = "preData",
  Required = "required",
```

Reply:
removed this entirely